### PR TITLE
Merge upstream

### PR DESCRIPTION
**Pull request by TomPridham**

### CODE REVIEW

File: src/xlsx.rs [hunk 1]
```diff
@@ -141,38 +162,6 @@ impl FromStr for CellErrorType {
     }
 }
 
-/// https://learn.microsoft.com/en-us/dotnet/api/documentformat.openxml.spreadsheet.sheetstatevalues?view=openxml-2.8.1
-#[derive(Debug, Clone, Copy)]
-enum SheetState {
```
Comment (TomPridham) [inline src/xlsx.rs:146]: calamine ended up supporting this, so i took their changes to avoid further merge conflicts

File: src/xlsx.rs [hunk 2]
```diff
@@ -368,19 +368,38 @@ impl<RS: Read + Seek> Xlsx<RS> {
                             _ => (),
                         }
                     }
-                    if !state.is_very_hidden() {
```
Comment (TomPridham) [inline src/xlsx.rs:371]: this change was different. we seem to completely skip very hidden sheets, where they include them but mark them as such

File: src/xlsx.rs [hunk 3]
```diff
@@ -904,41 +966,48 @@ where
...
+
+    let mut row_index = 0;
+    let mut col_index = 0;
+
     loop {
         buf.clear();
         match xml.read_event_into(&mut buf) {
-            Ok(Event::Start(r_element)) if r_element.local_name().as_ref() == b"row" => {
-                cols = 0;
+            Ok(Event::Start(ref row_element)) if row_element.local_name().as_ref() == b"row" => {
```
Comment (TomPridham) [inline src/xlsx.rs:978]: again, took calamines changes over ours

File: src/xlsx.rs [hunk 4]
```diff
@@ -1095,36 +1151,6 @@ fn is_custom_date_format(format: &str, custom_parser: Option<fn(&str) -> bool>)
     is_date || format.bytes().all(|c| b"mdyMDYhsHS-/.: \\".contains(&c))
 }
 
-fn is_builtin_date_format_id(id: &[u8]) -> bool {
```
Comment (TomPridham) [inline src/xlsx.rs:1098]: this mirrored another function in the `formats.rs` file

**Comment by kyle-mccarthy** [comment]
Due to the number of changes I can't really effectively review this. I'm of the opinion that we should go with the upstream changes, they are probably safe/battle tested.

**Review by kyle-mccarthy — APPROVED** [review]

**Pull request merged**